src/rules/typing/assumed_size.rs

```diff
@@ -0,0 +1,139 @@
+use crate::ast::FortitudeNode;
+use crate::settings::Settings;
+use crate::{ASTRule, Rule, Violation};
+use itertools::Itertools;
+use tree_sitter::Node;
+
+/// Rules for catching assumed size variables
+
+pub struct AssumedSize {}
+
+impl Rule for AssumedSize {
+    fn new(_settings: &Settings) -> Self {
+        AssumedSize {}
+    }
+
+    fn explain(&self) -> &'static str {
+        "
+        Assumed size dummy arguments declared with a star `*` as the size should be
+        avoided, and assumed shape with a colon `:` should be used instead. There are
+        several downsides to assumed size, the main one being that the compiler is
+        not able to determine the array bounds, so it is not possible to check for
+        array overruns or to use the array in whole-array expressions.
```

Comment:
Looks good! With the addition of the second paragraph and example code, I think this can now be shortened:

```suggestion
        Assumed size dummy arguments declared with a star `*` as the size should be
        avoided. There are several downsides to assumed size, the main one being
        that the compiler is not able to determine the array bounds, so it is not
        possible to check for array overruns or to use the array in whole-array 
        expressions.
```